src/component/plotly-dash-preview/render.js

```diff
@@ -0,0 +1,57 @@
+const remote = require('../../util/remote')
+
+/**
+ * @param {object} info : info object
+ *  - layoutType TODO
+ *  - direction TODO
+ *  - backgroundColor
+ *  - panels
+ * @param {object} opts : component options
+ * @param {function} sendToMain
+ *  - errorCode
+ *  - result
+ *    - imgData
+ */
+function render (info, opts, sendToMain) {
+  const PRINT_TO_PDF = (info.format === 'pdf')
+  const result = {}
+
+  let win = remote.createBrowserWindow({
+    width: info.width,
+    height: info.height
+  })
+
+  const contents = win.webContents
+  let errorCode = null
+
+  const done = () => {
+    win.close()
+
+    if (errorCode) {
+      result.msg = 'dash preview generation failed'
+    }
+    sendToMain(errorCode, result)
+  }
+
+  win.on('closed', () => {
+    win = null
+  })
+
+  contents.on('page-favicon-updated', () => {
```

Comment:
This is a placeholder for now .. !! It will be replaced with a custom event fired by Dash App once the rendering has finished.